app/services/password_hasher.rb

```diff
@@ -0,0 +1,24 @@
+class PasswordHasher
+  # IMPORTANT! changing COST will invalidate existing password hashes.
+  COST = if Rails.env.test?
+           '800$8$1$'.freeze # SCrypt::Engine.calibrate(max_time: 0.01)
+         else
+           '4000$8$4$'.freeze # SCrypt::Engine.calibrate(max_time: 0.5)
```

Comment:
ah that's tricky. Because of the `if Rails.env.test?` there is always going to be a code path that goes untested.

I have re-written to (hopefully) avoid that in 45679a0 which allowed me to address @jessieay comment about the code comments.